provider/ec2/environ.go

```diff
@@ -1890,7 +1892,9 @@ func (e *environ) destroyControllerManagedModels(ctx context.ProviderCallContext
 	}
 
 	roles, err := listRolesForController(ctx, e.iamClient, controllerUUID)
-	if err != nil {
+	if errors.IsUnauthorized(err) {
+		logger.Warningf("unable to roles for deletion, roles may have to be manually cleaned up for controller %q", controllerUUID)
```

Comment:
typo
also, can we include (if available) the controller name in the message (here and above)
eg
`... manually cleaned up for controller "foo" (uuid-dead-beef)`

Reply:
we don't have the controller name.